integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/PersonFhirResourceProviderIntegrationTest.java

```diff
@@ -0,0 +1,455 @@
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Bundle results = readBundleResponse(response);
+		
+		assertThat(results, notNullValue());
+		assertThat(results.getType(), equalTo(Bundle.BundleType.SEARCHSET));
+		assertThat(results.hasEntry(), is(true));
+		
+		List<Bundle.BundleEntryComponent> entries = results.getEntry();
+		
+		assertThat(entries, everyItem(hasProperty("fullUrl", startsWith("http://localhost/ws/fhir2/R3/Person/"))));
+		assertThat(entries, everyItem(hasResource(instanceOf(Person.class))));
+		assertThat(entries, everyItem(hasResource(validResource())));
+	}
+	
+	@Test
+	public void shouldReturnSortedAndFilteredSearchResultsForPersonsAsJson() throws Exception {
+		MockHttpServletResponse response = get("/Person?family=Doe&_sort=given").accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Bundle results = readBundleResponse(response);
+		
+		assertThat(results, notNullValue());
+		assertThat(results.getType(), equalTo(Bundle.BundleType.SEARCHSET));
+		assertThat(results.hasEntry(), is(true));
+		
+		List<Bundle.BundleEntryComponent> entries = results.getEntry();
+		
+		assertThat(entries, everyItem(hasResource(hasProperty("nameFirstRep", hasProperty("family", startsWith("Doe"))))));
+		assertThat(entries,
+		    containsInRelativeOrder(
+		        hasResource(hasProperty("nameFirstRep", hasProperty("givenAsSingleString", containsString("Jean")))),
+		        hasResource(hasProperty("nameFirstRep", hasProperty("givenAsSingleString", containsString("John"))))));
+		assertThat(entries, everyItem(hasResource(validResource())));
+	}
+	
+	@Test
+	public void shouldSearchForAllPersonsAsXML() throws Exception {
+		MockHttpServletResponse response = get("/Person").accept(FhirMediaTypes.XML).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Bundle results = readBundleResponse(response);
+		
+		assertThat(results, notNullValue());
+		assertThat(results.getType(), equalTo(Bundle.BundleType.SEARCHSET));
+		assertThat(results.hasEntry(), is(true));
+		
+		List<Bundle.BundleEntryComponent> entries = results.getEntry();
+		
+		assertThat(entries, everyItem(hasProperty("fullUrl", startsWith("http://localhost/ws/fhir2/R3/Person/"))));
+		assertThat(entries, everyItem(hasResource(instanceOf(Person.class))));
+		assertThat(entries, everyItem(hasResource(validResource())));
+	}
+	
+	@Test
+	public void shouldReturnSortedAndFilteredSearchResultsForPersonsAsXML() throws Exception {
+		MockHttpServletResponse response = get("/Person?family=Doe&_sort=given").accept(FhirMediaTypes.XML).go();
```

Comment:
You get an error for this test because "family" is not a configured endpoint for the Person resource. You might as well try: 
`get("/Person?name=Doe&_sort=given")`

Reply:
Hmmm... that's actually a bug in the endpoint. We should support all the standard name properties on Person too.

Reply:
@ibacher I don't think we can do that. That's because FHIR itself doesn't allow given and family for the person resource.
http://hl7.org/implement/standards/fhir/person.html

Reply:
I guess that's why they aren't there!... Well that's a bit silly, but ok.